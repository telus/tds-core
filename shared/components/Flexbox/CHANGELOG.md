# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.2](https://github.com/telus/tds-core/compare/@tds/shared-flexbox@1.0.1...@tds/shared-flexbox@1.0.2) (2019-02-04)

**Note:** Version bump only for package @tds/shared-flexbox

<a name="1.0.1"></a>

## 1.0.1 (2018-06-06)

**Note:** Version bump only for package @tds/shared-flexbox
