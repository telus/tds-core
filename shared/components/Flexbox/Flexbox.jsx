import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../utils/safeRest'
import joinClassNames from '../../utils/joinClassNames'
import capitalize from '../../utils/capitalize'

import styles from './Flexbox.modules.scss'

const Flexbox = ({ direction, justifyContent, dangerouslyAddClassName, children, ...rest }) => {
  const classNames = joinClassNames(
    styles[direction],
    justifyContent && styles[`justifyContent${capitalize(justifyContent)}`],
    dangerouslyAddClassName
  )

  return (
    <div {...safeRest(rest)} className={classNames}>
      {children}
    </div>
  )
}

Flexbox.propTypes = {
  direction: PropTypes.oneOf(['column', 'row']).isRequired,
  justifyContent: PropTypes.oneOf(['spaceBetween', 'center']),
  dangerouslyAddClassName: PropTypes.string,
  children: PropTypes.node.isRequired,
}

Flexbox.defaultProps = {
  justifyContent: undefined,
  dangerouslyAddClassName: undefined,
}

export default Flexbox
