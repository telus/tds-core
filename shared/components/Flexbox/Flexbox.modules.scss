.column {
  display: flex;
  flex-direction: column;
}

.row {
  display: flex;
  flex-direction: row;
}

.justifyContentSpaceBetween {
  justify-content: space-between;
}
