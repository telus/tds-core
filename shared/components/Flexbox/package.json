{
  "name": "@tds/shared-flexbox",
  "version": "1.0.0",
  "private": true,
  "main": "",
  "module": ""
}
