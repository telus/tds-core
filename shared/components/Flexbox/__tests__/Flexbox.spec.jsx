import React from 'react'
import { shallow } from 'enzyme'

import Flexbox from '../Flexbox'

describe('Flexbox', () => {
  const defaultProps = {
    direction: 'row',
  }
  const doShallow = (props = {}) =>
    shallow(
      <Flexbox {...defaultProps} {...props}>
        Some content
      </Flexbox>
    )

  it('renders', () => {
    const flexbox = doShallow()

    expect(flexbox).toMatchSnapshot()
  })

  it('can be a flex row or column', () => {
    let flexbox = doShallow({ direction: 'column' })
    expect(flexbox).toHaveClassName('column')

    flexbox = doShallow({ direction: 'row' })
    expect(flexbox).toHaveClassName('row')
  })

  it('can justify content', () => {
    const flexbox = doShallow({ justifyContent: 'spaceBetween' })

    expect(flexbox).toHaveClassName('justifyContentSpaceBetween')
  })

  it('will add additional arbitrary class names', () => {
    const flexbox = doShallow({ dangerouslyAddClassName: 'a-class' })

    expect(flexbox).toHaveClassName('a-class')
  })

  it('passes additional attributes to the HTML element', () => {
    const flexbox = doShallow({ id: 'the-id', 'data-some-property': 'a value' })

    expect(flexbox).toHaveProp('id', 'the-id')
    expect(flexbox).toHaveProp('data-some-property', 'a value')
  })

  it('does not allow custom CSS', () => {
    const flexbox = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(flexbox).not.toHaveProp('className', 'my-custom-class')
    expect(flexbox).not.toHaveProp('style')
  })
})
