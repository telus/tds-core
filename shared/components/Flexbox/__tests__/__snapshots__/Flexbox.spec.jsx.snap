// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flexbox renders 1`] = `
<div
  className="row"
>
  Some content
</div>
`;
